react-accordion-component/src/accordion.jsx

```diff
@@ -0,0 +1,46 @@
+import React from 'react';
+
+class Accordion extends React.Component {
+  constructor(props) {
+    super(props);
+    this.props = props;
+    this.state = {
+      target: null,
+      isTargetOn: false
+    };
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(e) {
+    !this.state.target
+      ? this.setState({ target: e.target.textContent, isTargetOn: !this.state.isTargetOn })
+      : this.setState(() => {
+        if (this.state.target !== e.target.textContent) {
+          return { target: e.target.textContent };
+        } else {
+          return { target: null, isTargetOn: !this.state.isTargetOn };
+        }
+
+      });
+  }
+
+  render() {
+    // if (this.state.isTitleClicked) {
+    return (
+      <div>
+        {this.props.list.map((item, index) => (
+          <div className="box" key={index} >
+            <h3 className="title" onClick={this.handleClick}>{item.subject}</h3>
```

Comment:
Here you can pass the index to `handleClick` in a closure.
```jsx
<h3 className="title" onClick={() => this.handleClick(index)}>{item.subject}</h3>
```

Reply:
Yes, yes! This is a great solution, I'd like to see better solution to improve the code efficiency.

Updated my solution:
809ed81